bundles/org.openhab.binding.mikrotik/src/main/java/org/openhab/binding/mikrotik/internal/handler/MikrotikInterfaceThingHandler.java

```diff
@@ -0,0 +1,283 @@
+
+import java.math.BigDecimal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.mikrotik.internal.MikrotikBindingConstants;
+import org.openhab.binding.mikrotik.internal.config.InterfaceThingConfig;
+import org.openhab.binding.mikrotik.internal.model.*;
+import org.openhab.binding.mikrotik.internal.util.RateCalculator;
+import org.openhab.binding.mikrotik.internal.util.StateUtil;
+import org.openhab.core.thing.*;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MikrotikInterfaceThingHandler} is a {@link MikrotikBaseThingHandler} subclass that wraps shared
+ * functionality for all interface things of different types. It is responsible for handling commands, which are
+ * sent to one of the channels and emit channel updates whenever required.
+ *
+ * @author Oleg Vivtash - Initial contribution
+ */
+@NonNullByDefault
+public class MikrotikInterfaceThingHandler extends MikrotikBaseThingHandler<InterfaceThingConfig> {
+    private final Logger logger = LoggerFactory.getLogger(MikrotikInterfaceThingHandler.class);
+
+    private @Nullable RouterosInterfaceBase iface;
+
+    private final RateCalculator txByteRate = new RateCalculator(BigDecimal.ZERO);
+    private final RateCalculator rxByteRate = new RateCalculator(BigDecimal.ZERO);
+    private final RateCalculator txPacketRate = new RateCalculator(BigDecimal.ZERO);
+    private final RateCalculator rxPacketRate = new RateCalculator(BigDecimal.ZERO);
+
+    public static boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return MikrotikBindingConstants.THING_TYPE_INTERFACE.equals(thingTypeUID);
+    }
+
+    public MikrotikInterfaceThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
+        if (getRouteros() != null) {
+            if (status == ONLINE || (status == OFFLINE && statusDetail == ThingStatusDetail.COMMUNICATION_ERROR)) {
+                getRouteros().registerForMonitoring(config.name);
+            } else if (status == OFFLINE && (statusDetail == ThingStatusDetail.CONFIGURATION_ERROR
+                    || statusDetail == ThingStatusDetail.GONE)) {
+                getRouteros().unregisterForMonitoring(config.name);
+            }
+        }
+        super.updateStatus(status, statusDetail, description);
+    }
+
+    @Override
+    protected void refreshModels() {
+        logger.trace("Searching for {} interface", config.name);
+        iface = getRouteros().findInterface(config.name);
+        if (iface == null) {
+            logger.warn("Interface {} is not found in RouterOS for thing {}", config.name, getThing().getUID());
+            updateStatus(OFFLINE, GONE, "Interface not found in RouterOS");
```

Comment:
The status detail message in `updateStatus()` is already logged by the framework. You could merge these two message.